### DISCUSSION TITLE
Allowing onDismiss for Toaster Toasts

### DISCUSSION THREAD

**Discussion by lupitadavila** [twilio-labs/paste#1802] [Ideas] [answered]
For the `Toaster` component, can we add `onDismiss` to the [useToaster toast](https://paste.twilio.design/components/toast/#usetoastertoast)? Or is there a reason for not having it there, but allowing it when creating a `Toast` component?

We'd like to be able to push a toast to the toaster and have a callback that saves that the toast was dismissed in the backend without it disrupting the other toasts in the array.

**Comment by SiTaggart** [comment]
Hi @lupitadavila 

Thanks for the question! You are absolutely spot on, this is totally an oversight on our behalf that we missed.

I'm thinking that you probably should just be able to add it to the object when you're pushing a toast to the toaster. Something like

```
const handleClick = () => {
  toaster.push({
    message: 'Selection was successfully saved to your profile',
    variant: 'success',
    onDismiss: () => {...}
  })
}
```

And all it is, is the same as `onDismiss` for an [individual Toast component](https://paste.twilio.design/components/toast/#toast-props)

Does that sound about right to you?

**Reply by lupitadavila** [reply]
Yes, that is the idea!

**Reply by lupitadavila** [reply]
If there is an estimate on timeline for us to track, that would be great. Thanks a bunch 🙏🏼

**Reply by TheSisb** [reply]
I'll try to get this fix in by early next week, if that works for you.

**Comment by TheSisb** [comment, marked as answer]
A PR is open here: https://github.com/twilio-labs/paste/pull/1818

**Reply by nkrantz** [reply]
@lupitadavila this PR has been merged!
